packages/wallet-manager/src/wallets/Evm/Evm.ts

```diff
@@ -0,0 +1,177 @@
+import { ethers } from 'ethers';
+import events from 'events';
+import type {
+  Web3Provider,
+  ExternalProvider,
+  Provider
+} from '@ethersproject/providers';
+import { IEvmWallet } from '../../interfaces';
+import { AddChain } from '../../types';
+import { checkWindow } from '../../utils';
+
+class EvmWallet extends events.EventEmitter implements IEvmWallet {
+  address?: string;
+  signer?: ethers.Signer;
+  web3Provider!: Web3Provider;
+
+  constructor(provider?: Web3Provider) {
+    super();
+
+    if (!window.ethereum) {
+      throw new Error('window.ethereum is not defined.');
+    }
+    if (provider) {
+      this.web3Provider = provider;
+    } else {
+      this.web3Provider = new ethers.providers.Web3Provider(
+        window.ethereum as ExternalProvider
+      );
+    }
+    this.appendProviderEvents();
+  }
+
+  /**
+   * @name calculateAccountData
+   * @param accounts
+   * @returns {Promise<void>}
+   */
+  private async calculateAccountData(accounts?: string[]): Promise<void> {
+    if (accounts?.length) {
+      this.address = accounts[0];
+    }
+
+    const _accounts = await this.web3Provider.listAccounts();
+
+    if (!_accounts.length) {
+      return;
+    }
+
+    this.address = _accounts![0];
+    this.signer = this.web3Provider.getSigner();
+  }
+
+  /**
+   * @name reconnectToProvider
+   * @returns {void}
+   */
+  private reconnectToProvider(): void {
+    this.web3Provider = new ethers.providers.Web3Provider(
+      window.ethereum as ExternalProvider
+    );
+  }
+
+  /**
+   * @name appendProviderEvents
+   * @returns {void}
+   * @description Appends the provider events to the windowConnector
+   */
+  private appendProviderEvents(): void {
+    try {
+      checkWindow();
+    } catch (e) {
+      throw e;
+    }
+    (this.web3Provider.provider as Provider).on('connect', async () => {
+      this.reconnectToProvider();
+      await this.calculateAccountData();
+    });
+
+    (this.web3Provider.provider as Provider).on(
+      'disconnect',
+      async (error: Error & { code: number; data?: unknown }) => {
```

Comment:
any reason for this being de-structured here? I don't see either of those really being used anywhere.

Reply:
no, since we removed the log